digdag-core/src/main/java/io/digdag/core/database/ThreadLocalTransactionManager.java

```diff
@@ -82,21 +140,37 @@ public void abort()
             if (handle == null) {
                 return;
             }
+            if (state == State.COMMITTED) {
+                throw new IllegalStateException("Aborting committed transaction is not allowed");
+            }
             handle.rollback();
+            state = State.ABORTED;
             // TODO Should call close()?
         }
     }
 
     @Inject
     public ThreadLocalTransactionManager(DataSource ds)
     {
+        this(ds, false);
+    }
+
+    ThreadLocalTransactionManager(DataSource ds, boolean autoAutoCommit)
+    {
         this.ds = checkNotNull(ds);
+        if (autoAutoCommit) {
+            LazyTransaction transaction = new LazyTransaction(ds, true);
```

Comment:
A purpose of autoAutoCommit is that threads don't have to call `begin()` or even `autoCommit()`. Instead, threads directly call `getHandle`. And it starts autoCommit transaction (which is not actually transaction though).
And `getHandle` is still lazy operation. So I think this is fine.